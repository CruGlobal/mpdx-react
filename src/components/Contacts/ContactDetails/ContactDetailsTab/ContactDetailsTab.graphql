query ContactDetailsTab($accountListId: ID!, $contactId: ID!) {
  contact(accountListId: $accountListId, id: $contactId) {
    id
    name
    primaryAddress {
      street
      city
      state
      postalCode
      country
    }
    tagList
    envelopeGreeting
    preferredContactMethod
    timezone
    user {
      firstName
      lastName
    }
    website
    sendNewsletter
    greeting
    #TODO add Magazine
    ...ContactPeople
  }
}
