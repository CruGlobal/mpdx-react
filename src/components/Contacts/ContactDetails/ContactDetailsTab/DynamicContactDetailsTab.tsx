import dynamic from 'next/dynamic';
import { DynamicComponentPlaceholder } from 'src/components/DynamicPlaceholders/DynamicComponentPlaceholder';

export const preloadContactDetailsTab = () =>
  import(
    /* webpackChunkName: "ContactDetailsTab" */ './ContactDetailsTab'
  ).then(({ ContactDetailsTab }) => ContactDetailsTab);

export const DynamicContactDetailsTab = dynamic(preloadContactDetailsTab, {
  loading: DynamicComponentPlaceholder,
});
