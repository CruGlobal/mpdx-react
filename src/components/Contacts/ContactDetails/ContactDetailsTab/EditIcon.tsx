import CreateIcon from '@mui/icons-material/Create';
import { styled } from '@mui/material/styles';

export const EditIcon = styled(CreateIcon)(({ theme }) => ({
  width: '18px',
  height: '18px',
  margin: theme.spacing(0),
  color: theme.palette.cruGrayMedium.main,
}));
