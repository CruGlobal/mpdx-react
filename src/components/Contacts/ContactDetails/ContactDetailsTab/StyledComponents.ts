import Add from '@mui/icons-material/Add';
import Create from '@mui/icons-material/Create';
import { Button, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';

export const AddIcon = styled(Add)(({ theme }) => ({
  color: theme.palette.info.main,
}));

export const EditIcon = styled(Create)(({ theme }) => ({
  width: '18px',
  height: '18px',
  margin: theme.spacing(0),
  color: theme.palette.cruGrayMedium.main,
}));

export const AddButton = styled(Button)(({ theme }) => ({
  margin: theme.spacing(1),
}));

export const AddText = styled(Typography)(({ theme }) => ({
  color: theme.palette.info.main,
  textTransform: 'uppercase',
  fontWeight: 'bold',
}));
