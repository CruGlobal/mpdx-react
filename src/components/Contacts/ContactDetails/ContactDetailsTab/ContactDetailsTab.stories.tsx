import React, { ReactElement } from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { CssBaseline } from '@mui/material';
import { ThemeProvider } from '@mui/material/styles';
import { GqlMockedProvider } from '__tests__/util/graphqlMocking';
import theme from '../../../../theme';
import { ContactDetailProvider } from '../ContactDetailContext';
import { ContactDetailsTab } from './ContactDetailsTab';
import { ContactDetailsTabDocument } from './ContactDetailsTab.generated';

export default {
  title: 'Contacts/Tab/ContactDetailsTab',
  component: ContactDetailsTab,
};

const accountListId = '111';
const contactId = '222';

export const Default = (): ReactElement => {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <GqlMockedProvider>
        <ContactDetailProvider>
          <ContactDetailsTab
            accountListId={accountListId}
            contactId={contactId}
            onContactSelected={() => {}}
          />
        </ContactDetailProvider>
      </GqlMockedProvider>
    </ThemeProvider>
  );
};

export const Loading = (): ReactElement => {
  return (
    <MockedProvider
      mocks={[
        {
          request: {
            query: ContactDetailsTabDocument,
            variables: {
              accountListId: accountListId,
              contactId: contactId,
            },
          },
          result: {},
          delay: 8640000,
        },
      ]}
    >
      <ContactDetailProvider>
        <ContactDetailsTab
          accountListId={accountListId}
          contactId={contactId}
          onContactSelected={() => {}}
        />
      </ContactDetailProvider>
    </MockedProvider>
  );
};
