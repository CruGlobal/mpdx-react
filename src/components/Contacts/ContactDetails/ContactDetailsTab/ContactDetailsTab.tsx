import React from 'react';
import { Box, Divider, Skeleton, Typography } from '@mui/material';
import { styled } from '@mui/material/styles';
import { useTranslation } from 'react-i18next';
import {
  ContactDetailContext,
  ContactDetailsType,
} from '../ContactDetailContext';
import { useContactDetailsTabQuery } from './ContactDetailsTab.generated';
import { ContactDetailsTabMailing } from './Mailing/ContactDetailsTabMailing';
import { ContactDetailsOther } from './Other/ContactDetailsOther';
import { EditContactOtherModal } from './Other/EditContactOtherModal/EditContactOtherModal';
import { ContactDetailsPartnerAccounts } from './PartnerAccounts/ContactDetailsPartnerAccounts';
import { ContactDetailsTabPeople } from './People/ContactDetailsTabPeople';
import { ContactTags } from './Tags/ContactTags';

const ContactDetailsTabContainer = styled(Box)(({ theme }) => ({
  width: '100%',
  padding: theme.spacing(0),
}));

const ContactDetailSectionContainer = styled(Box)(({ theme }) => ({
  padding: theme.spacing(0),
  paddingTop: '4px',
  margin: 0,
  alignContent: 'start',
}));

const ContactDetailHeadingContainer = styled(Box)(() => ({
  display: 'flex',
  alignItems: 'center',
}));

const ContactDetailHeadingText = styled(Typography)(() => ({
  flexGrow: 5,
  fontWeight: 'bold',
}));

const ContactDetailLoadingPlaceHolder = styled(Skeleton)(({ theme }) => ({
  width: '100%',
  height: '24px',
  margin: theme.spacing(2, 0),
}));

interface ContactDetailTabProps {
  accountListId: string;
  contactId: string;
  onContactSelected: (
    contactId: string,
    openDetails?: boolean,
    flows?: boolean,
  ) => void;
}

export const ContactDetailsTab: React.FC<ContactDetailTabProps> = ({
  accountListId,
  contactId,
  onContactSelected,
}) => {
  const { data, loading } = useContactDetailsTabQuery({
    variables: { accountListId, contactId },
  });

  const { t } = useTranslation();

  const { editOtherModalOpen, setEditOtherModalOpen } = React.useContext(
    ContactDetailContext,
  ) as ContactDetailsType;

  return (
    <>
      <ContactDetailsTabContainer>
        {
          // Tag Section
        }
        <ContactDetailHeadingContainer mt={-3}>
          {loading || !data ? (
            <ContactDetailLoadingPlaceHolder variant="rectangular" />
          ) : (
            <ContactTags
              accountListId={accountListId}
              contactId={data.contact.id}
              contactTags={data.contact.tagList}
            />
          )}
        </ContactDetailHeadingContainer>
        <Divider />
        {
          // People Section
        }
        <ContactDetailSectionContainer>
          {loading || !data ? (
            <>
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
            </>
          ) : (
            <ContactDetailsTabPeople
              data={data?.contact}
              accountListId={accountListId}
            />
          )}
        </ContactDetailSectionContainer>
        <Divider />
        {
          // Mailing Section
        }
        <ContactDetailSectionContainer>
          <ContactDetailHeadingContainer>
            <ContactDetailHeadingText variant="h6">
              {t('Mailing')}
            </ContactDetailHeadingText>
          </ContactDetailHeadingContainer>
          {loading || !data ? (
            <>
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
            </>
          ) : (
            <ContactDetailsTabMailing
              accountListId={accountListId}
              data={data.contact}
            />
          )}
        </ContactDetailSectionContainer>
        <Divider />
        {
          // other Section
        }
        <ContactDetailSectionContainer>
          <ContactDetailHeadingContainer>
            <ContactDetailHeadingText variant="h6">
              {t('Other')}
            </ContactDetailHeadingText>
          </ContactDetailHeadingContainer>
          {loading || !data ? (
            <>
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
            </>
          ) : (
            <ContactDetailsOther
              contact={data.contact}
              onContactSelected={onContactSelected}
              handleOpen={setEditOtherModalOpen}
            />
          )}
        </ContactDetailSectionContainer>
        <Divider />
        {
          // Patner Accounts Section
        }
        <ContactDetailSectionContainer>
          <ContactDetailHeadingContainer>
            <ContactDetailHeadingText variant="h6">
              {t('Partner Accounts')}
            </ContactDetailHeadingText>
          </ContactDetailHeadingContainer>
          {loading || !data ? (
            <>
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
              <ContactDetailLoadingPlaceHolder variant="rectangular" />
            </>
          ) : (
            <ContactDetailsPartnerAccounts contact={data.contact} />
          )}
        </ContactDetailSectionContainer>
        <Divider />
      </ContactDetailsTabContainer>
      {loading || !data ? null : (
        <EditContactOtherModal
          accountListId={accountListId}
          contact={data.contact}
          isOpen={editOtherModalOpen}
          referral={data.contact.contactReferralsToMe?.nodes[0]}
          handleClose={() => setEditOtherModalOpen(false)}
        />
      )}
    </>
  );
};
