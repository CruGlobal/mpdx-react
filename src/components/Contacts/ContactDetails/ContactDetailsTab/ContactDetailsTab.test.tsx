import { renderHook } from '@testing-library/react-hooks';
import { GqlMockedProvider } from '../../../../../__tests__/util/graphqlMocking';
import { useContactDetailsTabQuery } from './ContactDetailsTab.generated';

describe('ContactDetailTab', () => {
  it('test query', async () => {
    const { result, waitForNextUpdate } = renderHook(
      () =>
        useContactDetailsTabQuery({
          variables: {
            accountListId: 'accountList-id',
            contactId: 'contact-id',
          },
        }),
      { wrapper: GqlMockedProvider },
    );
    await waitForNextUpdate();
    expect(result.current.variables).toMatchSnapshot();
    expect(result.current.data.contact.name).toMatchSnapshot();
  });
});
