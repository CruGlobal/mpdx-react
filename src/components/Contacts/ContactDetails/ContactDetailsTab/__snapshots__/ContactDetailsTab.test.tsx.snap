// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactDetailTab test query 1`] = `
Object {
  "accountListId": "accountList-id",
  "contactId": "contact-id",
}
`;

exports[`ContactDetailTab test query 2`] = `"Elephant Film Stomach"`;
